components/org.wso2.carbon.identity.local.auth.push.authenticator/src/main/java/org/wso2/carbon/identity/local/auth/push/authenticator/PushAuthenticator.java

```diff
@@ -0,0 +1,1702 @@
+     */
+    protected AuthenticatorConstants.ScenarioTypes resolveScenario(HttpServletRequest request,
+                                                                             AuthenticationContext context) {
+
+        if (context.isLogoutRequest()) {
+            return LOGOUT;
+        } else if (StringUtils.isNotBlank(request.getParameter(SCENARIO)) &&
+                RESEND_PUSH_NOTIFICATION.getValue().equals(request.getParameter(SCENARIO))) {
+            return RESEND_PUSH_NOTIFICATION;
+        } else if (StringUtils.isNotBlank(request.getParameter(SCENARIO)) &&
+                PROCEED_PUSH_AUTHENTICATION.getValue().equals(request.getParameter(SCENARIO))) {
+            return PROCEED_PUSH_AUTHENTICATION;
+        } else if (StringUtils.isNotBlank(request.getParameter(SCENARIO)) &&
+                PUSH_DEVICE_ENROLLMENT.getValue().equals(request.getParameter(SCENARIO))) {
+            return PUSH_DEVICE_ENROLLMENT;
+        } else if (StringUtils.isNotBlank(request.getParameter(SCENARIO)) &&
+                INIT_PUSH_ENROLL.getValue().equals(request.getParameter(SCENARIO))) {
+            return INIT_PUSH_ENROLL;
+        } else if (StringUtils.isNotBlank(request.getParameter(SCENARIO)) &&
+                CANCEL_PUSH_ENROLL.getValue().equals(request.getParameter(SCENARIO))) {
+            return CANCEL_PUSH_ENROLL;
+        }
+        return SEND_PUSH_NOTIFICATION;
+    }
+
+    /**
+     * Get the number of maximum attempts the user is allowed resend the OTP.
+     *
+     * @return The maximum number of resend attempts.
+     * @throws AuthenticationFailedException If an error occurs when retrieving config.
+     */
+    protected int getMaximumResendAttempts(String tenantDomain) throws AuthenticationFailedException {
+
+        try {
+            return Integer.parseInt(AuthenticatorUtils.getPushAuthenticatorConfig(
+                    RESEND_NOTIFICATION_MAX_ATTEMPTS, tenantDomain));
+        } catch (PushAuthenticatorServerException e) {
+            throw handleAuthErrorScenario(ERROR_CODE_ERROR_GETTING_CONFIG);
+        }
+    }
+
+    /**
+     * Get the authenticated user from the context.
+     *
+     * @param context Authentication context.
+     * @return AuthenticatedUser.
+     * @throws AuthenticationFailedException Authentication failed exception.
+     */
+    private AuthenticatedUser getAuthenticatedUserFromContext(AuthenticationContext context)
+            throws AuthenticationFailedException {
+
+        Map<Integer, StepConfig> stepConfigMap = context.getSequenceConfig().getStepMap();
+        for (StepConfig stepConfig : stepConfigMap.values()) {
+            AuthenticatedUser user = stepConfig.getAuthenticatedUser();
+            if (stepConfig.isSubjectAttributeStep() && user != null) {
+                if (StringUtils.isBlank(user.toFullQualifiedUsername())) {
+                    LOG.debug("Username can not be empty.");
+                    throw handleAuthErrorScenario(ERROR_CODE_EMPTY_USERNAME);
```

Comment:
Why is both ERROR_CODE_EMPTY_USERNAME and ERROR_CODE_NO_USER_FOUND there? if not needed remove one.